lib/src/test/java/edu/stanford/futuredata/macrobase/analysis/classify/stats/NormalDistTest.java

```diff
@@ -0,0 +1,61 @@
+package edu.stanford.futuredata.macrobase.analysis.classify.stats;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+import org.apache.commons.math3.distribution.NormalDistribution;
+
+import java.io.*;
+import java.util.HashMap;
+import java.util.Map;
+
+public class NormalDistTest {
+    @Test
+    public void testCDF() throws Exception {
+        NormalDist dist = new NormalDist();
+        double cdf;
+
+        cdf = dist.cdf(0.0, 1.0, 0.5);
+        assertEquals(0.6915, cdf, 0.001);
+
+        cdf = dist.cdf(5.0, 2.0, 4.0);
+        assertEquals(0.3085, cdf, 0.001);
+
+        // Test interpolation
+        cdf = dist.cdf(0.0, 1.0, 0.755);
+        assertEquals(0.7749, cdf, 0.001);
+
+        // Test max z-score
+        cdf = dist.cdf(0.0, 1.0, 4.0);
+        assertEquals(1.0, cdf, 0.001);
+
+        // Test min z-score
+        cdf = dist.cdf(0.0, 1.0, -4.0);
+        assertEquals(0.0, cdf, 0.001);
+    }
+
+    @Test
+    public void constructLUT() {
```

Comment:
I would precompute it once and then just save the array statically. You can have a function in the WeightedPercentile class then that recomputes the array so you can compare the static value with the computed one in a unit test.